README.md
<h2>Version: 0.0.9</h2>

<h1>Usage:</h1>

<h3>Array</h3>

```ruby

array=["fishing", "bait", "worm"]

array.include? fish  # This is a built in Array function.
# => false

array.partial_include? fish
# => true

array.partial_include? shi
# => true

array.partial_include? shark
# => false

array=[{:foo=>"bar"}, {:bar=>"baz"}, {:baz=>"foo"}]

array.json_pretty
# => [
  {
    "foo": "bar"
  },
  {
    "bar": "baz"
  },
  {
    "baz": "foo"
  }
]

```

<h3>Hash</h3>

```ruby

hash={:one=>"un", :two=>"deux", :three=>"trois", :four=>"quatre"}

hash.json_pretty
# => {
  "one": "un",
  "two": "deux",
  "three": "trois",
  "four": "quatre"
}

```

<h3>Integer</h3>

```ruby

1234567890.to_human
# => 1.15GB

1234567890.to_human(5)
# => 1.14978GB

8549176320.to_human(1)
# => 8.0GB

12.to_human
# => 12B

5234566788764321.to_human
# => 4.65PB

1234567890987654321.to_human
# => 1.07EB

```

<h3>String</h3>

```ruby

string="TrUe"

string.to_b
# => true

"F".to_b
# => false

string="Television"

string.first
# => T

string.middle
# => elevisio

string.last
# => n

string.shuffle
# => eveosiliTn

string.scramble
# => Tileesoivn

string.valid_json?
# => false

string="jOhN sMiTh"

string.proper
# => John Smith

string="nretnil kram"

string.proper
# => Nretnil Kram

string="One two 3."

string.proper
# => One Two 3.

string.first
# => O

string.middle
# => ne two 3

string.last
# => .

string.shuffle
# => 3O.e  notw

string.scramble
# => Oo3netw  .

string.valid_json?
# => false

string='{ "One Octet" }'

string.valid_json?
# => false

string='{ "foo": "bar" }'

string.valid_json?
# => true

string='7*d8f7R6$5H@~32'

string.sym?
# => true

string='78d8f7R6k5H4g2'

string.sym?
# => false

string.uuid?
# => false

string='0b5865e9-0262-6238-1427-35b77c466ebd'

string.uuid?
# => true

string='0865e9-026-6238-127-35b77c46bd'

string.uuid?
# => false

string='xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'

string.uuid?
# => false

```

<h3>Time</h3>

```ruby

timing
# => 2018-04-03 15:00:30 -0400

timing(12.8345679)
# => 13 seconds

timing(123)
# => 02 minutes 03 seconds

timing(1_234.56789)
# => 20 minutes 35 seconds

timing(12_345)
# => 03 hours 25 minutes 45 seconds

```

<h3>Utilities</h3>

```ruby

6.times do
  fifty_fifty
end
# => true
# => false
# => true
# => true
# => true
# => false

6.times do
  one_third
end
# => false
# => false
# => true
# => false
# => false
# => true

6.times do
  coin_flip
end
# => tails'
# => tails'
# => 'heads
# => 'heads
# => 'heads
# => 'heads

```


CHANGELOG.md
<h4>0.0.9</h4>

* added json_pretty to Array

<h4>0.0.8</h4>

* added one_third function

<h4>0.0.7</h4>

* added fifty_fifty and coin_flip functions

<h4>0.0.6</h4>

* added timing function

<h4>0.0.5</h4>

* added string.proper

<h4>0.0.4</h4>

* added string.sym? and string.uuid?

<h4>0.0.3</h4>

* rubocop changes

<h4>0.0.2</h4>

* Added Integer.to_human for file sizes

<h4>0.0.1</h4>

* Initial Gem Build


examples/example.rb
#!/bin/env ruby
# frozen_string_literal: true

require 'rubygems'
require 'nretnil-utilities'

puts '<h3>Array</h3>'

puts "\n"

puts '```ruby'
puts "\n"

array = %w[fishing bait worm]

puts "array=#{array}\n\n"

puts 'array.include? fish  # This is a built in Array function.'
puts '# => ' + array.include?('fish').to_s
puts "\n"

puts 'array.partial_include? fish'
puts '# => ' + array.partial_include?('fish').to_s
puts "\n"

puts 'array.partial_include? shi'
puts '# => ' + array.partial_include?('shi').to_s
puts "\n"

puts 'array.partial_include? shark'
puts '# => ' + array.partial_include?('shark').to_s
puts "\n"

array = [{ foo: 'bar' }, { bar: 'baz' }, { baz: 'foo' }]

puts "array=#{array}\n\n"

puts 'array.json_pretty'
puts '# => ' + array.json_pretty
puts "\n"

puts '```'
puts "\n"

puts '<h3>Hash</h3>'

puts "\n"

puts '```ruby'
puts "\n"

hash = { one: 'un', two: 'deux', three: 'trois', four: 'quatre' }

puts "hash=#{hash}\n\n"

puts 'hash.json_pretty'
puts '# => ' + hash.json_pretty.to_s
puts "\n"

puts '```'
puts "\n"

puts '<h3>Integer</h3>'

puts "\n"

puts '```ruby'
puts "\n"

puts '1234567890.to_human'
puts '# => ' + 1_234_567_890.to_human
puts "\n"

puts '1234567890.to_human(5)'
puts '# => ' + 1_234_567_890.to_human(5)
puts "\n"

puts '8549176320.to_human(1)'
puts '# => ' + 8_549_176_320.to_human(1)
puts "\n"

puts '12.to_human'
puts '# => ' + 12.to_human
puts "\n"

puts '5234566788764321.to_human'
puts '# => ' + 5_234_566_788_764_321.to_human
puts "\n"

puts '1234567890987654321.to_human'
puts '# => ' + 1_234_567_890_987_654_321.to_human
puts "\n"

puts '```'
puts "\n"

puts '<h3>String</h3>'

puts "\n"

puts '```ruby'
puts "\n"

string = 'TrUe'

puts "string=\"#{string}\"\n\n"

puts 'string.to_b'
puts '# => ' + string.to_b.to_s
puts "\n"

puts '"F".to_b'
puts '# => ' + 'F'.to_b.to_s
puts "\n"

string = 'Television'

puts "string=\"#{string}\"\n\n"

puts 'string.first'
puts '# => ' + string.first
puts "\n"

puts 'string.middle'
puts '# => ' + string.middle
puts "\n"

puts 'string.last'
puts '# => ' + string.last
puts "\n"

puts 'string.shuffle'
puts '# => ' + string.shuffle
puts "\n"

puts 'string.scramble'
puts '# => ' + string.scramble
puts "\n"

puts 'string.valid_json?'
puts '# => ' + string.valid_json?.to_s
puts "\n"

string = 'jOhN sMiTh'

puts "string=\"#{string}\"\n\n"

puts 'string.proper'
puts '# => ' + string.proper
puts "\n"

string = 'nretnil kram'

puts "string=\"#{string}\"\n\n"

puts 'string.proper'
puts '# => ' + string.proper
puts "\n"

string = 'One two 3.'

puts "string=\"#{string}\"\n\n"

puts 'string.proper'
puts '# => ' + string.proper
puts "\n"

puts 'string.first'
puts '# => ' + string.first
puts "\n"

puts 'string.middle'
puts '# => ' + string.middle
puts "\n"

puts 'string.last'
puts '# => ' + string.last
puts "\n"

puts 'string.shuffle'
puts '# => ' + string.shuffle
puts "\n"

puts 'string.scramble'
puts '# => ' + string.scramble
puts "\n"

puts 'string.valid_json?'
puts '# => ' + string.valid_json?.to_s
puts "\n"

string = '{ "One Octet" }'

puts "string=\'#{string}\'\n\n"

puts 'string.valid_json?'
puts '# => ' + string.valid_json?.to_s
puts "\n"

string = '{ "foo": "bar" }'

puts "string=\'#{string}\'\n\n"

puts 'string.valid_json?'
puts '# => ' + string.valid_json?.to_s
puts "\n"

string = '7*d8f7R6$5H@~32'

puts "string=\'#{string}\'\n\n"

puts 'string.sym?'
puts '# => ' + string.sym?.to_s
puts "\n"

string = '78d8f7R6k5H4g2'

puts "string=\'#{string}\'\n\n"

puts 'string.sym?'
puts '# => ' + string.sym?.to_s
puts "\n"

puts 'string.uuid?'
puts '# => ' + string.uuid?.to_s
puts "\n"

string = '0b5865e9-0262-6238-1427-35b77c466ebd'

puts "string=\'#{string}\'\n\n"

puts 'string.uuid?'
puts '# => ' + string.uuid?.to_s
puts "\n"

string = '0865e9-026-6238-127-35b77c46bd'

puts "string=\'#{string}\'\n\n"

puts 'string.uuid?'
puts '# => ' + string.uuid?.to_s
puts "\n"

string = 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'

puts "string=\'#{string}\'\n\n"

puts 'string.uuid?'
puts '# => ' + string.uuid?.to_s
puts "\n"

puts '```'
puts "\n"

puts '<h3>Time</h3>'

puts "\n"

puts '```ruby'
puts "\n"

puts 'timing'
puts '# => ' + timing.to_s
puts "\n"

puts 'timing(12.8345679)'
puts '# => ' + timing(12.8345679)
puts "\n"

puts 'timing(123)'
puts '# => ' + timing(123)
puts "\n"

puts 'timing(1_234.56789)'
puts '# => ' + timing(1_234.56789)
puts "\n"

puts 'timing(12_345)'
puts '# => ' + timing(12_345)
puts "\n"

puts '```'
puts "\n"

puts '<h3>Utilities</h3>'

puts "\n"

puts '```ruby'
puts "\n"

puts '6.times do'
puts '  fifty_fifty'
puts 'end'
6.times do
  puts '# => ' + fifty_fifty.to_s
end
puts "\n"

puts '6.times do'
puts '  one_third'
puts 'end'
6.times do
  puts '# => ' + one_third.to_s
end
puts "\n"

puts '6.times do'
puts '  coin_flip'
puts 'end'
6.times do
  puts '# => ' + coin_flip
end
puts "\n"

puts '```'
puts "\n"
